pytube/__main__.py

```diff
@@ -102,6 +103,34 @@ def __init__(
             self.prefetch()
             self.descramble()
 
+    def check_availability(self):
+        """Check whether the video is available.
+        Raises different exceptions based on why the video is unavailable,
+        otherwise does nothing.
+
+        """
+        if self.watch_html is None:
+            raise VideoUnavailable(video_id=self.video_id)
+
+        status, messages = extract.playability_status(self.watch_html)
+        for reason in messages:
+            if status == 'UNPLAYABLE':
+                if reason == (
+                    'Join this channel to get access to members-only content '
+                    'like this video, and other exclusive perks.'
+                ):
+                    raise MembersOnly(video_id=self.video_id)
+                elif reason == 'This live stream recording is not available.':
```

Comment:
This won't work if the IP is not from an English speaking country!

Reply:
See above